queries.sql

```diff
@@ -8,3 +9,45 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name NOT IN ('Gabumon');
 SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <=17.3;
+
+/* Milestone 2*/
+
+-- Transaction 1 
+BEGIN;
+UPDATE animals SET species='unspecified';
+ROLLBACK;
+
+-- Transaction 2
+BEGIN;
+UPDATE animals SET species='digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species='pokemon' WHERE species IS NULL;
+COMMIT;
+
+-- Transaction 3
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+
+-- Transaction 4
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT SP1;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO SP1;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+
+-- Queries
+
+-- How many animals are there?
+SELECT COUNT(*) FROM animals;
+-- How many animals have never tried to escape?
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+-- What is the average weight of animals?
+SELECT AVG(weight_kg) FROM animals;
+-- Who escapes the most, neutered or not neutered animals?
+SELECT SUM(escape_attempts), neutered FROM animals GROUP BY neutered;
+-- What is the minimum and maximum weight of each type of animal?
+SELECT MIN(weight_kg), MAX(weight_kg), species from animals GROUP BY species;
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+SELECT AVG(escape_attempts) FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31';
```

Comment:
- [ ] Your result should be groped by animal type, you can add `GROUP BY species` so your query can be like: 
```
 SELECT species, AVG(escape_attempts) FROM animals
   GROUP BY species
   WHERE  date_of_birth >= '1990/01/01' AND date_of_birth <= '2000/12/31';;
```

Reply:
Thanks, I've just done it